features/_trustedboot/file.include/usr/lib/dracut/modules.d/99sysroot-loop/module-setup.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+check() {
+	return 0
+}
+
+depends() {
+	echo "systemd"
+	return 0
+}
+
+install() {
+	mkdir -p "$initdir/etc/systemd/system"
+
+	cat > "$initdir/etc/systemd/system/sysroot.mount" <<-EOF
+	[Mount]
+	What=/root.img
+	Where=/sysroot
+	Type=erofs
+	EOF
+
+	mkdir -p "$initdir/etc/systemd/system/initrd-root-fs.target.requires"
+	ln -s "$initdir/etc/systemd/system/sysroot.mount" "$initdir/etc/systemd/system/initrd-root-fs.target.requires/"
+
+	inst /root.img /root.img
+	instmods erofs
```

Comment:
Just as a logic, should we not start with a modprobe first?

Reply:
that should all be handled by dracut, this is not code that is actually run on the target system, just the logic dracut uses to generate the initrd. instmods here is some dracut specific magic, dracut handles all the details for us.